tests/components/azure_event_hub/test_config_flow.py

```diff
@@ -0,0 +1,198 @@
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CS_CONFIG_FULL,
+        title="test-instance",
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": source},
+        data=BASE_CONFIG_CS.copy(),
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+@pytest.mark.parametrize(
+    "side_effect, error_message",
+    [(EventHubError("test"), "cannot_connect"), (Exception, "unknown")],
+    ids=["cannot_connect", "unknown"],
+)
+async def test_connection_error_sas(
+    hass,
+    mock_get_eventhub_properties,
+    side_effect,
+    error_message,
+):
+    """Test we handle connection errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data=BASE_CONFIG_SAS.copy(),
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+    mock_get_eventhub_properties.side_effect = side_effect
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        SAS_CONFIG.copy(),
+    )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": error_message}
+
+
+@pytest.mark.parametrize(
+    "side_effect, error_message",
+    [(EventHubError("test"), "cannot_connect"), (Exception, "unknown")],
+    ids=["cannot_connect", "unknown"],
+)
+async def test_connection_error_cs(
+    hass,
+    mock_from_connection_string,
+    side_effect,
+    error_message,
+):
+    """Test we handle connection errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data=BASE_CONFIG_CS.copy(),
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+    mock_from_connection_string.return_value.get_eventhub_properties.side_effect = (
+        side_effect
+    )
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        CS_CONFIG.copy(),
+    )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": error_message}
+
+
+async def test_options_flow(hass, entry):
+    """Test options flow."""
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "options"
+    assert result["last_step"]
+
+    updated = await hass.config_entries.options.async_configure(
+        result["flow_id"], UPDATE_OPTIONS
+    )
+    assert updated["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert updated["data"] == UPDATE_OPTIONS
+    await hass.async_block_till_done()
+
+    assert (
+        hass.data[DOMAIN][DATA_HUB]._send_interval  # pylint: disable=protected-access
```

Comment:
In these config flow tests, only assert that the options in the config entry has been updated.

If we want to test that the interval etc has changed for the hub, we test this in the init module tests by moving time forward and checking that the send calls happen at the expected interval.

https://github.com/home-assistant/core/blob/da7c5f4722d5b10309c88941128f8adde97866ad/tests/common.py#L372-L376

Reply:
I;ve now added a mock for the update_options func, this lets me check it was called and a seperate test in test_init to check if a call to update_options results in updated values, that oughta do it!